.template/addons/github/.github/workflows/deploy_aws_fargate.yml.tt

```diff
@@ -0,0 +1,50 @@
+name: Deploy to AWS Fargate
+
+on:
+  workflow_run:
+    workflows:
+      - Test
+    branches:
+      - main
+    types:
+      - completed
+  workflow_dispatch:
+
+env:
+  DOCKER_REGISTRY_HOST: ${{ secrets.DOCKER_REGISTRY_HOST }}
+  DOCKER_IMAGE: ${{ secrets.ECR_REPO }}
+  ECS_CLUSTER: ${{ secrets.ECS_CLUSTER }}
+  ECS_SERVICE: ${{ secrets.ECS_SERVICE }}
+  BRANCH_TAG: latest
+
+jobs:
+  deploy:
+    name: Deploy to AWS Fargate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - uses: actions/checkout@v2.3.4
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build Docker image
+        run: bin/docker-prepare && docker-compose build
+
+      - name: Push to ECR
+        run: docker push $DOCKER_REGISTRY_HOST/$DOCKER_IMAGE:$BRANCH_TAG
```

Comment:
Since you've used `docker-compose build` to build, should we use `docker-compose push` here as well?

Reply:
this doesn't work because [docker-compose push](https://docs.docker.com/compose/reference/push/) doesn't support pushing to a private registry unless specifying in the docker-compose file, e.g. `ECR`, but [docker push](https://docs.docker.com/engine/reference/commandline/push/#push-a-new-image-to-a-registry) does